src/components/Header/Heading.js

```diff
@@ -39,22 +40,63 @@ const levels = [
   margins.withComponent("h5")
 ];
 
-const Heading = ({ level, children, ...props }) => {
+const Heading = ({
+  level,
+  size,
+  responsiveSize,
+  lineHeight,
+  children,
+  ...props
+}) => {
   const H = levels[level - 1];
-  return <H {...props}>{children}</H>;
+  return (
+    <H
+      {...props}
+      size={{
+        small: responsiveSize.small || size || "tera",
+        medium:
+          responsiveSize.medium || responsiveSize.small || size || "zetta",
+        large:
+          responsiveSize.large ||
+          responsiveSize.medium ||
+          responsiveSize.small ||
+          size ||
+          "bronto"
+      }}
+      lineHeight={lineHeight || "body"}
+    >
+      {children}
+    </H>
+  );
 };
 
 Heading.propTypes = {
   level: PropTypes.oneOf([1, 2, 3, 4, 5]),
+  size: PropTypes.oneOf([Object.keys(typography.size)]),
+  responsiveSize: PropTypes.shape({
+    small: PropTypes.oneOf([Object.keys(typography.size)]),
+    medium: PropTypes.oneOf([Object.keys(typography.size)]),
+    large: PropTypes.oneOf([Object.keys(typography.size)])
+  }),
+  lineHeight: PropTypes.oneOf([Object.keys(typography.lineHeight)]),
   children: PropTypes.node
 };
 
 Heading.defaultProps = {
   level: 2,
+  size: null,
+  responsiveSize: {
+    small: null,
+    medium: null,
+    large: null
+  },
+  lineHeight: null,
```

Comment:
Maybe default the `lineHeight` prop to `"body"` and remove the inline or operator above?